dartagnan/src/main/java/com/dat3m/dartagnan/wmm/relation/base/RelCrit.java

```diff
@@ -29,6 +29,7 @@ public TupleSet getMinTupleSet(){
         return minTupleSet;
     }
 
+	// This is an over-approximation of the real rcu-rscs since we don't do proper matching. 
```

Comment:
What matching are we missing?

Reply:
`RelCrit` basically adds every pair where the first event is a `rcu_lock` and the second is a `rcu_unlock` if the cid of the first is smaller. For this program
```
e1: rcu_lock
...
e2: rcu_unlock
e3: rcu_lock
...
e4: rcu_unlock
```
we would add `(e1,e4)` while (AFAIK) herd would not. However these extra pairs do not seem to cause problems in our current test suite.

Reply:
Well, the `maxTupleSet` might be unnecessarily large, but the encoding does make sure that `(e1, e4)` are never matched, so semantically we are precise. It's just that the `maxTupleSet` is a little too large.

Reply:
You are right. I hadn't check the implementation of `encodeApprox` but that one take care of doing the proper matching